File: c/retangulo.c
#include <stdio.h>
#include <math.h>

int main()
{
    double x, y, a, p, d;

    printf("Base do retangulo: ");
    scanf("%lf", &x);

    printf("Altura do retangulo: ");
    scanf("%lf", &y);

    a = x * y;

    p = x * 2 + y * 2;

    d =  sqrt(pow(x, 2.0) + pow(y, 2.0));


    printf("AREA = %.4lf\n", a);
    printf("PERIMETRO = %.4lf\n", p);
    printf("DIAGONAL = %.4lf\n", d);
}

File: csharp/retangulo.cs
using System;
using System.Globalization;

namespace teste
{
    internal class Program
    {
        static void Main(string[] args)
        {

            CultureInfo CI = CultureInfo.InvariantCulture;

            double baseX, alturaX, areaX, perimetroX, diagonalX;

            Console.Write("Base do retangulo: ");
            baseX = double.Parse(Console.ReadLine(), CI);

            Console.Write("Altura do retangulo: ");
            alturaX = double.Parse(Console.ReadLine(), CI);

            areaX = alturaX * baseX;
            Console.WriteLine("AREA = " + areaX.ToString("F4", CI));

            perimetroX = 2 * (baseX + alturaX);
            Console.WriteLine("PERIMETRO = " + perimetroX.ToString("F4", CI));

            diagonalX = Math.Sqrt(Math.Pow(alturaX, 2.0) +  Math.Pow(baseX, 2.0));
            Console.WriteLine("DIAGONAL = " + diagonalX.ToString("F4", CI));
        }
    }
}
